File: multiplication_without_star.py
def multiplication_without_star(a,b,multiplicator=None):

    if multiplicator == None:
        # swap a and b, when the function runs first time, for optimization
        if b>a:
            a = a+b
            b = a-b
            a = a-b
        multiplicator = a

    # base case
    if b == 1:
        return a
    # recursive case
    else:
        a += multiplicator
        return multiplication_without_star(a,b-1,multiplicator)
    
print(multiplication_without_star(3,7))

File: multiplication_without_star.js
const multiplication_without_star = (a,b,multiplicator) => {
    if (multiplicator == undefined){
        // swap a and b, when function runs for first time, for optimization
        a = a + b;
        b = a - b;
        a = a - b;
        //
        multiplicator = a;
    }
    
    // base case
    if (b == 1){
        return a;
    }
    else{
        a += multiplicator;
        return multiplication_without_star(a,b-1,multiplicator);
    }
}

console.log(multiplication_without_star(6,2));

// Time complexity: O(n)
// Space complexity: O(n)